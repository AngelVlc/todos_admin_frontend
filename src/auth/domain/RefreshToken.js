export class RefreshToken {
  constructor({id, userId, expirationDate}) {
    this._id = id;
    this._userId = userId;
    this._expirationDate = expirationDate;
    this._selected = false;
  }

  get id() {
    return this._id;
  }

  get userId() {
    return this._userId;
  }

  get expirationDate() {
    return this._expirationDate;
  }

  get selected() {
    return this._selected;
  }

  set selected(value) {
    this._selected = value;
  } 
}