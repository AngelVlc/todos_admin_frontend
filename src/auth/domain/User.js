export class User {
  constructor({ id, name, isAdmin }) {
    this.id = id;
    this.name = name;
    this.isAdmin = isAdmin;
    this.password = "";
    this.confirmPassword = "";
  }

  static createEmpty() {
    return new User({ name: "", isAdmin: false });
  }
}
