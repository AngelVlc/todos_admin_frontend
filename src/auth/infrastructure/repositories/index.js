export * from './RefreshTokensRepository';
export * from './UsersRepository';
