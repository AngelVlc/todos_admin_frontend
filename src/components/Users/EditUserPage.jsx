import React, { useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';
import { UserForm } from './UserForm';
import axios from 'axios';

export const EditUserPage = () => {
    let { userId } = useParams();

    const [pageState, setPageState] = useState({name: ''});

    useEffect(() => {
        const getExistingUser = async () => {
            const res = await axios.get(`users/${userId}`)
            const user = res.data

            setPageState({
                name: user.name,
                isAdmin: user.isAdmin ? 'yes' : 'no'
            });
        }
        getExistingUser();
    }, [userId]);

    return (
        <div className="container">
            <h3 className="title">{`Edit user '${pageState.name}'`}</h3>
            <nav className="breadcrumb" aria-label="breadcrumbs">
                <ul>
                    <li><Link to={`/`}>Home</Link></li>
                    <li><Link to={`/users`}>Users</Link></li>
                    <li className="is-active"><Link aria-current="page" to={`/users/${userId}`}>{pageState.name}</Link></li>
                </ul>
            </nav>
            <UserForm userId={userId} name={pageState.name} isAdmin={pageState.isAdmin} isNew={false} submintBtnText='SAVE' submitUrl={`users/${userId}`}></UserForm>
        </div>
    )
}