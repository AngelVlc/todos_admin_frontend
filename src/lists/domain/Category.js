export class Category {
  constructor({ id, name, description }) {
    this.id = id;
    this.name = name;
    this.description = description;
  }

  static createEmpty() {
    return new Category({ id: -1, name: "", description: "" });
  }
}
