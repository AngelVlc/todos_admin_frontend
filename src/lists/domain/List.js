export class List {
  constructor({ id, name, itemsCount, items = [] }) {
    this.id = id;
    this.name = name;
    this.items = items;
    this.itemsCount = itemsCount;
  }

  static createEmpty() {
    return new List({ id: -1, name: "", itemsCount: 0 });
  }

  addNewItem(newItem) {
    newItem.id = -1;

    for (const item of this.items) {
      if (item.id <= newItem.id) {
        newItem.id = item.id - 1;
      }
    };

    this.items.push(newItem);
  }
}
