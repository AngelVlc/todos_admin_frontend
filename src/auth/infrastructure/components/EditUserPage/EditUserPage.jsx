import React, { useEffect, useState, useContext, useCallback } from "react";
import { useParams, useHistory } from "react-router-dom";
import { Breadcrumb } from "../../../../shared/infrastructure/components/Breadcrumb";
import { UserForm } from "../UserForm";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { GetUserByIdUseCase } from "../../../application/users";

export const EditUserPage = () => {
  let history = useHistory();
  let { userId } = useParams();
  const { useCaseFactory } = useContext(AppContext);
  const [pageState, setUser] = useState();

  const getUser = useCallback(async () => {
    const getUserByIdUseCase = useCaseFactory.get(GetUserByIdUseCase);
    const data = await getUserByIdUseCase.execute(userId);

    setUser(data);
  }, [userId, useCaseFactory]);

  useEffect(() => {
    getUser();
  }, [userId, getUser]);

  return (
    <>
      {pageState && (
        <div className="container">
          <Breadcrumb
            items={[
              { url: "/users", text: "Users" },
              { url: `/users/${userId}`, text: pageState.name },
            ]}
          />
          <h3 className="title">{`Edit user '${pageState.name}'`}</h3>
          <UserForm user={pageState}>
            <div className="control">
              <button
                className="button is-danger"
                data-testid="delete"
                type="button"
                onClick={() => history.push(`/users/${userId}/delete`)}
              >
                DELETE
              </button>
            </div>
          </UserForm>
        </div>
      )}
    </>
  );
};
