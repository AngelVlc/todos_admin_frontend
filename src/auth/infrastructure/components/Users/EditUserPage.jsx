import React, { useEffect, useState, useContext, useCallback } from "react";
import { Link, useParams } from "react-router-dom";
import { UserForm } from "./UserForm";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { GetUserByIdUseCase } from "../../../application/users";

export const EditUserPage = () => {
  let { userId } = useParams();
  const { useCaseFactory } = useContext(AppContext);
  const [pageState, setUser] = useState({ name: "" });

  const getUser = useCallback(async () => {
    const getUserByIdUseCase = useCaseFactory.get(GetUserByIdUseCase);
    const data = await getUserByIdUseCase.execute(userId);

    setUser(data);
  }, [userId, useCaseFactory]);

  useEffect(() => {
    getUser();
  }, [userId, getUser]);

  return (
    <div className="container">
      <nav className="breadcrumb" aria-label="breadcrumbs">
        <ul>
          <li>
            <Link to={`/`}>Home</Link>
          </li>
          <li>
            <Link to={`/users`}>Users</Link>
          </li>
          <li className="is-active">
            <Link aria-current="page" to={`/users/${userId}`}>
              {pageState.name}
            </Link>
          </li>
        </ul>
      </nav>
      <h3 className="title">{`Edit user '${pageState.name}'`}</h3>
      <UserForm user={pageState}/>
    </div>
  );
};
