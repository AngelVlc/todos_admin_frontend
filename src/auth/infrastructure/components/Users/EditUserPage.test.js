import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { EditUserPage } from "./EditUserPage";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { MemoryRouter, Route } from "react-router-dom";
import { act } from "react-dom/test-utils";
import { User } from "../../../domain";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const renderWithContextAndRouterForExistingUser = (isAdmin) => {
  const mockedGetUserByIdUseCase = {
    execute: jest.fn(),
  };

  const useCaseFactory = {
    get: () => {
      return mockedGetUserByIdUseCase;
    },
  };

  mockedGetUserByIdUseCase.execute.mockResolvedValue(
    new User({ id: 2, name: "user", isAdmin: isAdmin })
  );

  const context = { auth: { info: {} }, useCaseFactory };
  return {
    ...render(
      <AppContext.Provider value={context}>
        <MemoryRouter initialEntries={[`/users/2/edit`]}>
          <Route path="/users/:userId/edit">{<EditUserPage />}</Route>
        </MemoryRouter>
      </AppContext.Provider>
    ),
  };
};

afterEach(cleanup);

it("should match the snapshot for an existing non admin user", async () => {
  let fragment;
  await act(async () => {
    const { asFragment } = renderWithContextAndRouterForExistingUser(false);
    fragment = asFragment;
  });
  expect(fragment()).toMatchSnapshot();
});

it("should match the snapshot for an existing admin user", async () => {
  let fragment;
  await act(async () => {
    const { asFragment } = renderWithContextAndRouterForExistingUser(true);
    fragment = asFragment;
  });
  expect(fragment()).toMatchSnapshot();
});

it("should allow delete an existing user", async () => {
  let container;
  await act(async () => {
    container = renderWithContextAndRouterForExistingUser(false);
  });

  await waitFor(() => {
    fireEvent.click(container.getByTestId("delete"));
  });

  expect(mockHistoryPush.mock.calls.length).toBe(1);
  expect(mockHistoryPush.mock.calls[0][0]).toBe("/users/2/delete");
  mockHistoryPush.mockClear();
});
