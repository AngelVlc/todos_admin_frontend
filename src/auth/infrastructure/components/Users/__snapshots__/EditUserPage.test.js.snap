// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for an existing admin user 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/users"
          >
            Users
          </a>
        </li>
        <li
          class="is-active"
        >
          <a
            aria-current="page"
            href="/users/2"
          >
            user
          </a>
        </li>
      </ul>
    </nav>
    <h3
      class="title"
    >
      Edit user 'user'
    </h3>
    <form
      action="#"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="name"
        >
          Name
        </label>
        <div
          class="control"
        >
          <input
            data-testid="name"
            name="name"
            type="text"
            value="user"
          />
        </div>
        <p
          class="help is-danger"
          data-testid="userNameErrors"
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <div
          class="control"
        >
          <input
            data-testid="password"
            name="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="confirmPassword"
        >
          Confirm Password
        </label>
        <div
          class="control"
        >
          <input
            data-testid="confirmPassword"
            name="confirmPassword"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="isAdmin"
        >
          Is Admin
        </label>
        <div
          class="control"
        >
          <input
            data-testid="isAdmin"
            name="isAdmin"
            type="checkbox"
            value="true"
          />
        </div>
      </div>
      <div
        class="field is-grouped"
      >
        <p
          class="control"
        >
          <button
            class="button"
            data-testid="submit"
            type="submit"
          >
            UPDATE
          </button>
        </p>
        <p
          class="control"
        >
          <button
            class="button"
            data-testid="cancel"
            type="button"
          >
            CANCEL
          </button>
        </p>
        <p
          class="control"
        >
          <button
            class="button is-danger"
            data-testid="delete"
            type="button"
          >
            DELETE
          </button>
        </p>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot for an existing non admin user 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/users"
          >
            Users
          </a>
        </li>
        <li
          class="is-active"
        >
          <a
            aria-current="page"
            href="/users/2"
          >
            user
          </a>
        </li>
      </ul>
    </nav>
    <h3
      class="title"
    >
      Edit user 'user'
    </h3>
    <form
      action="#"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="name"
        >
          Name
        </label>
        <div
          class="control"
        >
          <input
            data-testid="name"
            name="name"
            type="text"
            value="user"
          />
        </div>
        <p
          class="help is-danger"
          data-testid="userNameErrors"
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <div
          class="control"
        >
          <input
            data-testid="password"
            name="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="confirmPassword"
        >
          Confirm Password
        </label>
        <div
          class="control"
        >
          <input
            data-testid="confirmPassword"
            name="confirmPassword"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="isAdmin"
        >
          Is Admin
        </label>
        <div
          class="control"
        >
          <input
            data-testid="isAdmin"
            name="isAdmin"
            type="checkbox"
            value="false"
          />
        </div>
      </div>
      <div
        class="field is-grouped"
      >
        <p
          class="control"
        >
          <button
            class="button"
            data-testid="submit"
            type="submit"
          >
            UPDATE
          </button>
        </p>
        <p
          class="control"
        >
          <button
            class="button"
            data-testid="cancel"
            type="button"
          >
            CANCEL
          </button>
        </p>
        <p
          class="control"
        >
          <button
            class="button is-danger"
            data-testid="delete"
            type="button"
          >
            DELETE
          </button>
        </p>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
